### DISCUSSION TITLE
What is the purpose of the NextUIProvider?

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [heroui-inc/heroui#1856] [Q&A] [answered]
Hi!

First of.. what an awesome lib 🥳!

I'm trying to understand the purpose of NextUIProvider.
I have searched the code and the docs but cant find any good information about it.

I'm doing a components lib with storybook and then going to use this on our NextJs project.
But i have not done anything with NextUIProvider and all components work as expected..

**Comment by jrgarciadev** [comment, marked as answer]
Hey @J4v4Scr1pt the `NextUIProvider` serves to control the global `locale` for language and `RTL` and for the Routing feature https://nextui.org/docs/guide/routing

**Reply by J4v4Scr1pt** [reply]
Thx 4 the answer! 

But then it's not needed if these features are not used I guess, or do you recommend always wrapping when using NextUI?

**Comment by martinGM05** [comment]
I have a question, If I want show the locale with es because I'm from Mexico but in the DatePicker just show me on the time field without de 'AM' - 'PM', just show it with en
![image](https://github.com/nextui-org/nextui/assets/63875704/e8eaea9e-4fc0-4c2f-9bbc-6a7339980e77)
